### PR TITLE
SampleGen: Remove newline at the end of a comment block

### LINKED ISSUE #2578: SampleGen: `comment` adds extraneous comment newline
My YAML:

```yaml
 on_success:
  - loop:
      variable: result
      collection: $resp.results
      body:
      - comment:
          - |
              First recognition hypothesis.
              These alternatives are ordered in terms of accuracy, with the top (first) alternative being the most probable.
      - comment:
          - "%s is the most probable recognition result."
          - first_alternative
      - define: first_alternative=result.alternatives[0]
# ...
```

My Python output:

```py
for result in response.results:
    # First recognition hypothesis.
    # These alternatives are ordered in terms of accuracy, with the top (first) alternative being the most probable.
    #
    # first_alternative is the most probable recognition result.
    first_alternative = result.alternatives[0]
```

The same is true in PHP:

```php
foreach ($response->getResults() as $result) {
    // First recognition hypothesis.
    // These alternatives are ordered in terms of accuracy, with the top (first) alternative being the most probable.
    //
    // firstAlternative is the most probable recognition result.
    $firstAlternative = $result->getAlternatives()[0];
```

As you can see, I needed a second `comment:` directive to make use of `%s` language-specific variable casing.

But I did not intend to add an empty comment line between the 2 `comment:` directives

=> @hzyi-google

### PR DESCRIPTION
**Pull request by hzyi-google**
Fix #2578

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/OutputTransformer.java [hunk 1]
```diff
@@ -397,7 +399,31 @@ private OutputView loopView(
...
+    // comment: |
+    //   I am line1
+    //   I am line2
+    // ```
+    // Parsed config would be: `I am line1\nI am line2\n`.
+    //
+    // And splittedComments would be: ["I am line1", "I am line2", ""]
+    //
+    // But we don't want the last empty string.
+    String[] splittedComments = formattedComment.split("\\n", -1);
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/transformer/OutputTransformer.java]: s/`splittedComments`/`splitComments`/g 
("split" is the past participle of "to split")

File: src/main/java/com/google/api/codegen/transformer/OutputTransformer.java [hunk 2]
```diff
@@ -397,7 +399,31 @@ private OutputView loopView(
             .map(c -> context.getNamer().localVarName(Name.anyLower(c)))
             .toArray(Object[]::new);
     String formattedComment = String.format(comment, args);
-    ImmutableList<String> lines = ImmutableList.copyOf(formattedComment.split("\\n", -1));
+    // We need to call split with a negative limit because user
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/transformer/OutputTransformer.java]: nit: "the user"

File: src/main/java/com/google/api/codegen/transformer/OutputTransformer.java [hunk 3]
```diff
@@ -397,7 +399,31 @@ private OutputView loopView(
...
+    // might explicitly specify empty comment lines at the end of
+    // the comment block.
+    // But we need to explicitly trim the last element because we don't
+    // want to include the blankline introduced by yaml syntax
+    //
+    // For example:
+    // If yaml looks like:
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/transformer/OutputTransformer.java]: nit: "the YAML"

File: src/main/java/com/google/api/codegen/transformer/OutputTransformer.java [hunk 4]
```diff
@@ -397,7 +399,31 @@ private OutputView loopView(
...
+    // ```
+    // comment: |
+    //   I am line1
+    //   I am line2
+    // ```
+    // Parsed config would be: `I am line1\nI am line2\n`.
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/transformer/OutputTransformer.java]: nit: let's have this read as one sentence

```
then the parsed ... \n`                <-- no period at end

and splitComments....

but we....
```

File: src/main/java/com/google/api/codegen/transformer/OutputTransformer.java [hunk 5]
```diff
@@ -397,7 +399,31 @@ private OutputView loopView(
             .map(c -> context.getNamer().localVarName(Name.anyLower(c)))
             .toArray(Object[]::new);
     String formattedComment = String.format(comment, args);
-    ImmutableList<String> lines = ImmutableList.copyOf(formattedComment.split("\\n", -1));
+    // We need to call split with a negative limit because user
+    // might explicitly specify empty comment lines at the end of
+    // the comment block.
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/transformer/OutputTransformer.java]: nit: make one sentence: "... comment block, but we need ...."

File: src/main/java/com/google/api/codegen/transformer/OutputTransformer.java (hunk 1 above)
Comment (hzyi-google) [reply src/main/java/com/google/api/codegen/transformer/OutputTransformer.java]: And I was hesitating between one 't' and two 't's...

**Pull request merged**